src/brod_client.erl

```diff
@@ -243,14 +250,13 @@ handle_info({'EXIT', Pid, Reason}, #state{ client_id     = ClientId
                                          } = State) ->
   error_logger:error_msg("client ~p producers supervisor down~nReason: ~p",
                          [ClientId, Pid, Reason]),
-  %% shutdown all producers?
-  {noreply, State#state{producers_sup = ?undef}};
+  {stop, {producers_sup_down, Reason}, State};
```

Comment:
I see no reason to have client running when producers_sup or consumers_sup died. State is lost anyway.